src/parser/parse.ts

```diff
@@ -0,0 +1,204 @@
+            resultType: 'error',
+            message: 'expected at least one value after "="',
+        }
+    } else if (values.length === 1) {
+        const fieldValue = tokenValueToFieldValue(values[0])
+        if (fieldValue !== null) {
+            return {
+                resultType: 'success',
+                value: {
+                    fieldValue,
+                    newPos: newPos,
+                },
+            }
+        } else {
+            return {
+                resultType: 'error',
+                message: `invalid value after the "=": ${tokensToString(
+                    values,
+                )}`,
+            }
+        }
+    } else {
+        const fieldValues: Array<TextField | IdentifierField> = []
+        for (const token of values) {
+            const value = tokenValueToFieldValue(token)
+            if (value !== null) {
+                fieldValues.push(value)
+            }
+        }
+        if (fieldValues.length === values.length) {
+            return {
+                resultType: 'success',
+                value: {
+                    fieldValue: {
+                        fieldType: 'list',
+                        list: fieldValues,
+                    },
+                    newPos: newPos,
+                },
+            }
+        } else {
+            return {
+                resultType: 'error',
+                message: `invalid values after the "=": ${tokensToString(
+                    values,
+                )}`,
+            }
+        }
+    }
+}
+
+/**
+ * Parse a field of the iPKG file.
+ *
+ * E.g: `main = Idris.Main`
+ */
+function parseField(
+    tokens: Array<UsefulToken>,
+    pos: number,
+): ParseResult<{ field: Field; newPos: number }> {
+    var newPos = pos
+    let ident = tokens[newPos]
+    let equals = tokens[newPos + 1]
+    newPos += 2
+
+    if (
+        ident &&
+        ident.tokenType === 'ident' &&
+        equals &&
+        equals.tokenType === 'equals'
+    ) {
+        const fieldValueResult = parseFieldValue(tokens, newPos)
+        if (fieldValueResult.resultType === 'success') {
+            const { newPos, fieldValue } = fieldValueResult.value
+            const field: Field = { name: ident.identifier, value: fieldValue }
+            return {
+                resultType: 'success',
+                value: {
+                    field,
+                    newPos,
+                },
+            }
+        } else {
+            return {
+                resultType: 'error',
+                message: `Couldn't parse the field value for ${ident.identifier}. Error: ${fieldValueResult.message}`,
+            }
+        }
+    } else {
+        return {
+            resultType: 'error',
+            message: `Expected an identifier followed by "=", but got: ${tokensToString(
+                [ident, equals],
+            )}`,
+        }
+    }
+}
+
+export function parsePackage(tokens: Array<UsefulToken>): ParseResult<Package> {
```

Comment:
The main/entry function of this file should be commented, no? At least to make it clear that this is the top-level-thingy in here. Maybe the `export` makes this totally obvious?